src/zarr/storage/_zip.py

```diff
@@ -108,7 +108,8 @@ async def _open(self) -> None:
         self._sync_open()
 
     def __getstate__(self) -> dict[str, Any]:
-        state = self.__dict__
+        # We need a copy to not modify the state of the original store
+        state = self.__dict__.copy()
```

Comment:
I'm not an expert on these methods, so I'd appreciate if someone with more experience could verify that we need a copy. Maybe @martindurant could help?

Reply:
Meaning, is there a better way to approach this than the current implementation?

Reply:
Since the `__init__` doesn't take many arguments, you could pluck the ones you need, and make this a `__reduce__` - slightly simpler. It means deciding which arguments you yes need, rather than the `pop` statements here deciding which ones you don't.

But yes, you need a copy() in order not to mutate the state of the original object.

Reply:
thank you!

Reply:
I looked into using `__reduce__` but find the current implementation to be simpler given that `ZipStore` requires kwargs, so I would need to use most of the optional tuple values returned by `__reduce__`. So, I'll leave this as is. thanks again for the advice